### PR TITLE
makes mounted guns easy to use

### PR DESCRIPTION
**Pull request by tyeagg**
## About The Pull Request
this puts the person manning the gun in the same turf of the mounted gun
you can also reload without having to get off it now removed clicking to get off gun since that just puts you inside of the tile of the gun with this use wasd to get off also unable to push gun if its manned 

atgun has a higher chance to block projectiles because shield

https://user-images.githubusercontent.com/98577688/191786171-e3c6806f-a6b5-4962-99b4-a7025d48b7fe.mp4

## Why It's Good For The Game
fixes issue of camera being 1 tile short of where your aiming since your in the same tile of the gun, you can turn 360 without a care in the world which would help in really tight spaces, nobody can move the gun if the gun is movable if your manning it, removes density from mounted gun when someone is manning it and reapplies it again when they get off this will help with the gun acting as a shield for the user like boiler glob can now direct hit user instead of the gun and also makes it so user can get hit by xeno abilities since their in the same tile as the gun

## Changelog
:cl:
qol: fixes issue of camera being 1 tile short of where your aiming since your in the same tile of the gun
qol: you can turn 360 in mounted gun without a care about obstacle blocking you
qol: nobody can move the gun if the gun is movable and if your manning it
qol: reload without getting off amounted gun
balance: mounted guns will no longer act as a shield if your manning it
balance: mounted guns now use coverage instead of completely stopping a projectile if it hits it
/:cl:

### CODE REVIEW

**Comment by FosterGrime-Heymynameslouie** [comment]
finally, no more fatass marines blocking tadpole hsg from turning around


speed merhg instantly

**Comment by maikilangiolo** [comment]
rip, cannot watch and laugh as 50 huggers bounce off my gun. Naturally, this is great qol. TAT is gonna be biting the dust vs boilers since you will be direct hit.

**Comment by Oscari-pog** [comment]
On one hand, not a shield anymore. On the other, aside not getting pushed off, itll allow weapons to be placed in more spots. By the way, if someone is on the HMG 08, which is movable, will someone be able to move them with them on it, or is that not how it works?

File: code/modules/projectiles/mounted.dm [hunk 1]
```diff
@@ -141,20 +144,43 @@
...
+	user_old_x = operator.pixel_x
+	user_old_y = operator.pixel_y
+	update_pixels(operator, TRUE)
+	density = FALSE
 	user_old_move_resist = operator.move_resist
 	operator.move_resist = MOVE_FORCE_STRONG
 
+///Updates the pixel offset of user so it looks like their manning the gun from behind
+/obj/machinery/deployable/mounted/proc/update_pixels(mob/user, mounting)
+	if(mounting)
```
Comment (BraveMole) [inline code/modules/projectiles/mounted.dm:158]: Use early return

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Comment by comfyorange** [comment]
#### Test Merge Deployed

##### Server Instance
TGMC

##### Merged By
Lewdcifer

##### Revision
Origin: 35d91d65c70e178a0573266fc54ab1d8d286bcf7
Pull Request: eecfe4421ff5abaab75d9bf45bdf4737e8dc05b2
Server: eab27903d70383cce7ec372016cd9398eab809b1

##### Comment

[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by tyeagg** [comment]
> This makes the user immune to xeno projectiles/getting pounced, apparently.

Projectile yea it hits the gun instead for some reason every other ability no

**Comment by Technical-l** [comment]
im pretty sure you can still fire the gun even while you are stunned now
got shot to shit by an HMG after i screeched them yesterday

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Lewdcifer

##### Revision
Origin: e3409a458c7646f94667b86b4118a2ddefe029be
Pull Request: eecfe4421ff5abaab75d9bf45bdf4737e8dc05b2
Server: 0cb0114322ab5bff59011db8ad0dde81bec04f0e

##### Comment

[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Lewdcifer

##### Revision
Origin: 0a6bcd46aec7f644f8feb2d1a994d8165e1bbb91
Pull Request: eecfe4421ff5abaab75d9bf45bdf4737e8dc05b2
Server: b101d8efcdad4319a5d4f1caffb0cf6a69621100

##### Comment

[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Review by Pariah919 — COMMENTED** [review]
Idea seems good to me, even if I'll miss trolling boilers with the gun shield.
Could make some of the guns apply a defense boost for the future.

**Comment by Lumipharon** [comment]
@tyeagg I assume it's this PR, otherwise no one has noticed this bug before, which is also likely.

If you get stunned while on a mounted gun firing, you can keep firing until the next tick. So that can be 1-8 bullets or so of brrt after I am horizontal.

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
naaanii

##### Revision
Origin: af73cb5a187ee4479e42fb50efc36e8fbcfdfc30
Pull Request: eecfe4421ff5abaab75d9bf45bdf4737e8dc05b2
Server: a82a7b9a2e34a0152938b45c435209b017c35eac

##### Comment

[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Tiviplus

##### Revision
Origin: af73cb5a187ee4479e42fb50efc36e8fbcfdfc30
Pull Request: 6804324b363e426124e645e69cf888c279d2fef9
Server: 2592483a876e876dcb90a7aa0c54b0fa595184c4

**Comment by FosterGrime-Heymynameslouie** [comment]
unsure if this is intended or not but xenos can shuffle ya at out manning the gun

also makes you drop on the floor whenever tad/alamo lands unlike before when you can have a chair and man a gun and you'll be fine

**Comment by tyeagg** [comment]
> unsure if this is intended or not but xenos can shuffle ya at out manning the gun
> 
> also makes you drop on the floor whenever tad/alamo lands unlike before when you can have a chair and man a gun and you'll be fine

intended

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Lewdcifer

##### Revision
Origin: af73cb5a187ee4479e42fb50efc36e8fbcfdfc30
Pull Request: 6804324b363e426124e645e69cf888c279d2fef9
Server: 99de39021678299d1ff1c63ce1c4f70ed66a98d9

##### Comment

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Tiviplus

##### Revision
Origin: 1a0d30d9b48151b309e3ce11f7ab98481c77c9a9
Pull Request: 8815ce1ae8d1dfbd0e35f6b7ac576bd0569bf353
Server: 7adad29020eeb4be86efb0ddff82c4d7eb2491b2

##### Comment

[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Lewdcifer

##### Revision
Origin: ba11e493e0e54387094b1aee3f723c962e51dc58
Pull Request: 8815ce1ae8d1dfbd0e35f6b7ac576bd0569bf353
Server: 1c1e4b98bed5b49a08101b5d08554feba05f2f0b

##### Comment

[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Lewdcifer

##### Revision
Origin: 74ddc18f3425d2a813310b95d0a2e0f96909bce9
Pull Request: 8815ce1ae8d1dfbd0e35f6b7ac576bd0569bf353
Server: 0630093ff9a7e29366983fccc1d30b0735259a83

##### Comment

**Comment by comfyorange** [comment]
#### Test Merge Removed

**Comment by FosterGrime-Heymynameslouie** [comment]
I hope this dosent to into the stale grave yard as do most of tyes cool pr goes to

holding on for a hero

File: code/modules/projectiles/mounted.dm [hunk 2]
```diff
@@ -7,6 +7,11 @@
 	layer = TANK_BARREL_LAYER
 	use_power = FALSE
 	hud_possible = list(MACHINE_HEALTH_HUD, MACHINE_AMMO_HUD)
+	flags_atom = ON_BORDER
+	throwpass = TRUE
+	///For animation and rotation of mounted guns
+	var/user_old_x = 0
+	var/user_old_y = 0
```
Comment (Lumipharon) [inline code/modules/projectiles/mounted.dm:14]: autodoc

File: code/modules/projectiles/projectile.dm [hunk 3]
```diff
@@ -693,11 +693,13 @@ So if we are on the 32th absolute pixel coordinate we are on tile 1, but if we a
 	return TRUE
 
 /obj/machinery/deployable/mounted/projectile_hit(obj/projectile/proj, cardinal_move, uncrossing)
+	. = ..()
+	var/hit_chance = coverage
+	hit_chance = min(hit_chance , hit_chance + 100 - proj.accuracy)
+	if(!density)
+		return prob(hit_chance)
```
Comment (Lumipharon) [inline code/modules/projectiles/projectile.dm:700]: doesn't this mean the IFF functionality for not hitting deployables stops working when mounted?
i.e. put this check after the iff if statement

**Review by Lumipharon — CHANGES_REQUESTED** [review]

**Comment by MereWalker** [comment]
Please oh please make this into the game. This is probably one of the best QOL things to do. Getting pushed around by your own team is the worst.

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**